src/main.rs

```diff
@@ -0,0 +1,272 @@
+    };
+    let key = HS256Key::from_bytes(key.as_bytes());
+    let claims =
+        Claims::create(Duration::from_hours(TOKEN_EXPIRE_HOURES)).with_subject("omnect-ui");
+
+    let Ok(token) = key.authenticate(claims) else {
+        error!("refresh_token: cannot create token");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+
+    HttpResponse::Ok().body(token)
+}
+
+async fn reboot(auth: BearerAuth) -> impl Responder {
+    debug!("reboot() called");
+
+    post("/reboot/v1", Some(auth)).await
+}
+
+async fn reload_network(auth: BearerAuth) -> impl Responder {
+    debug!("reload_network() called");
+
+    post("/reload-network/v1", Some(auth)).await
+}
+
+async fn post(path: &str, auth: Option<BearerAuth>) -> HttpResponse {
+    if let Some(auth) = auth {
+        let (status_code, error_msg) = verify(auth);
+
+        if status_code != StatusCode::OK {
+            error!("put {path} verify: {error_msg}");
+            return HttpResponse::build(status_code).finish();
+        }
+    }
+
+    let Ok(stream) =
+        UnixStream::connect(std::env::var("SOCKET_PATH").expect("SOCKET_PATH missing")).await
+    else {
+        error!("cannot create unix stream");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let Ok((mut sender, conn)) = http1::handshake(TokioIo::new(stream)).await else {
+        error!("unix stream handshake failed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+
+    actix_rt::spawn(async move {
+        if let Err(err) = conn.await {
+            error!("Connection failed1: {:?}", err);
+        }
+    });
+
+    if sender.ready().await.is_err() {
+        error!("unix stream unexpectedly closed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    }
+
+    let Ok(request) = Request::builder()
+        .uri(path)
+        .method("POST")
+        .header("Host", "localhost")
+        .body(Empty::<Bytes>::new())
+    else {
+        error!("build request failed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+
+    let Ok(res) = sender.send_request(request).await else {
+        error!("send request failed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let Ok(status_code) = StatusCode::from_u16(res.status().as_u16()) else {
+        error!("get status code failed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let Ok(body) = res.collect().await else {
+        error!("collect response body failed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
+    let Ok(body) = String::from_utf8(body.to_bytes().to_vec()) else {
+        error!("get response body failed");
+        return HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR).finish();
+    };
```

Comment:
Nit: could probably be refactored into separate function so that we only have a single place in code that catches any errors and handles them w/ HttpResponse::build(StatusCode::INTERNAL_SERVER_ERROR), instead of 5